### ISSUE TITLE
Cinder Kubernetes lock support

### ISSUE THREAD

**Issue by ricolin** [vexxhost/atmosphere#601]
Currently Cinder volume service only runs with single replica with file as coordinate backend.

Let's add kubernetes lock support with tooz so cinder can use it directly and allow us to do 3 replicas instead of 1.

**Comment by mpiscaer** [comment]
When the k8s node fails, isn't k8s who handle this single node to failover/recreate so that there is a minimum downtime?

If i understand it correctly, the a multinode setup is for scalability.

**Comment by ricolin** [comment]
@mpiscaer yeah, this is exactly for scalability purposes. k8s handling failover nicely that allow us do single volume service.
With this fix in tooz and openstack helm side, we should be able to scale and reach better experience with service failover.

**Comment by ricolin** [comment]
community patch: https://review.opendev.org/c/openstack/tooz/+/897561

**Comment by ricolin** [comment]
https://review.opendev.org/c/openstack/cinder/+/956528
https://github.com/vexxhost/atmosphere/pull/2851

**Comment by mnaser** [comment]
@ricolin Unfortunately, it seems that [Sherlock](https://github.com/py-sherlock/sherlock) is abandoned (3+ years with no changes).

What about creating a new project called `tooz-kubernetes` and then we can simply have it add an extra entry point and we can just add it to our images?

edit: the only annoying thing is that the `kubernetes` backend name is already used by something that probably doesn't work that well now.
